diff --git a/config.def.h b/config.def.h
index 78693b3..78f283e 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,16 +1,34 @@
 /* See LICENSE file for copyright and license details. */
 
+/* Solarized theme */
+#define SOLARIZED_BASE03    "#002b36"
+#define SOLARIZED_BASE02    "#073642"
+#define SOLARIZED_BASE01    "#586e75"
+#define SOLARIZED_BASE00    "#657b83"
+#define SOLARIZED_BASE0     "#839496"
+#define SOLARIZED_BASE1     "#93a1a1"
+#define SOLARIZED_BASE2     "#eee8d5"
+#define SOLARIZED_BASE3     "#fdf6e3"
+#define SOLARIZED_YELLOW    "#b58900"
+#define SOLARIZED_ORANGE    "#cb4b16"
+#define SOLARIZED_RED       "#dc322f"
+#define SOLARIZED_MAGNETA   "#d33682"
+#define SOLARIZED_VIOLET    "#6c71c4"
+#define SOLARIZED_BLUE      "#268bd2"
+#define SOLARIZED_CYAN      "#2aa198"
+#define SOLARIZED_GREEN     "#859900"
+
 /* appearance */
 static const char *fonts[] = {
 	"monospace:size=10"
 };
 static const char dmenufont[]       = "monospace:size=10";
-static const char normbordercolor[] = "#444444";
-static const char normbgcolor[]     = "#222222";
-static const char normfgcolor[]     = "#bbbbbb";
-static const char selbordercolor[]  = "#005577";
-static const char selbgcolor[]      = "#005577";
-static const char selfgcolor[]      = "#eeeeee";
+static const char normbordercolor[] = SOLARIZED_BASE01;
+static const char normbgcolor[]     = SOLARIZED_BASE03;
+static const char normfgcolor[]     = SOLARIZED_BASE01;
+static const char selbordercolor[]  = SOLARIZED_BASE2;
+static const char selbgcolor[]      = SOLARIZED_BASE03;
+static const char selfgcolor[]      = SOLARIZED_BASE2;
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const Bool showbar           = True;     /* False means no bar */
